tools/check_version.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+
+import sys
+from typing import List, Callable
+from ctypes import c_uint8, c_uint16
+import argparse
+import re
+
+#
+# MARK: - Standards
+#
+
+class Standard:
+    def __init__(self, is_version_compliant:Callable[[str], bool], success_message:str, error_message:str):
+        self.is_version_compliant:Callable[[str], bool] = is_version_compliant
+        self.success_message:str = success_message
+        self.error_message:str = error_message
+
+# Source: https://semver.org/
+semver_regex = "^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
+semver_standard = Standard(is_version_compliant=lambda version: re.match(semver_regex, version) is not(None),
```

Comment:
I would create a function `version_is_semver_compliant` here instead of using a lambda, for consistency with what comes next and to make it clearer to the reader